jota/src/main/java/org/iota/jota/model/Transaction.java

```diff
@@ -499,59 +409,167 @@ public String toTrytes() {
+        private String tag;
+        private long attachmentTimestampLowerBound;
+        private long attachmentTimestampUpperBound;
+
+        /**
+         * Start builder with curl mode CURL_P81 as default
+         */
+        public Builder() {
+            this.customCurl = SpongeFactory.create(SpongeFactory.Mode.CURL_P81);
+        }
+
+        public Builder curl(ICurl curl) {
+            this.customCurl = curl;
+            return this;
+        }
+
+        public Builder hash(String hash) {
+            this.hash = hash;
+            return this;
+        }
+
+        public Builder signatureFragments(String signatureFragments) {
+            this.signatureFragments = signatureFragments;
+            return this;
+        }
+
+        public Builder address(String address) {
+            this.address = address;
+            return this;
+        }
+
+        public Builder value(long value) {
+            this.value = value;
+            return this;
+        }
+
+        public Builder obsoleteTag(String obsoleteTag) {
+            this.obsoleteTag = obsoleteTag;
+            return this;
+        }
+
+        public Builder timestamp(long timestamp) {
+            this.timestamp = timestamp;
+            return this;
+        }
 
-        if (StringUtils.isEmpty(trytes)) {
-            log.warn("Warning: empty trytes in input for transactionObject");
-            return;
+        public Builder currentIndex(long currentIndex) {
+            this.currentIndex = currentIndex;
+            return this;
         }
 
-        // validity check
-        if (!InputValidator.isNinesTrytes(trytes.substring(2279, 2295), 16)) {
-            log.warn("Trytes {} does not seem a valid tryte", trytes);
-            return;
+        public Builder lastIndex(long lastIndex) {
+            this.lastIndex = lastIndex;
+            return this;
+        }
+
+        public Builder bundle(String bundle) {
+            this.bundle = bundle;
+            return this;
+        }
+
+        public Builder trunkTransaction(String trunkTransaction) {
+            this.trunkTransaction = trunkTransaction;
+            return this;
+        }
+
+        public Builder branchTransaction(String branchTransaction) {
+            this.branchTransaction = branchTransaction;
+            return this;
+        }
+
+        public Builder nonce(String nonce) {
+            this.nonce = nonce;
+            return this;
+        }
+
+        public Builder persistence(boolean persistence) {
+            this.persistence = persistence;
+            return this;
+        }
+
+        public Builder tag(String tag) {
+            this.tag = tag;
+            return this;
+        }
+
+        public Builder attachmentTimestamp(long attachmentTimestamp) {
+            this.attachmentTimestamp = attachmentTimestamp;
+            return this;
+        }
+
+        public Builder attachmentTimestampLowerBound(long attachmentTimestampLowerBound) {
```

Comment:
Codacy found an issue: [Public method and constructor comments are required](https://app.codacy.com/manual/Iota-java/iota-java/pullRequest?prid=6319461)